data/campaigns/fri04.wmf/scripting/mission_thread.lua

```diff
@@ -0,0 +1,173 @@
+local seen_reebaud = false
+local seen_amazons = false
+
+local o_defeat_amz = nil
+local o_defeat_emp = nil
+local amazons_defeated_by_reebaud = false
+
+function hurry_up()
+   while true do
+      sleep(10000)
+      if seen_reebaud or p4.defeated then return end
+      if #p2:get_buildings("frisians_headquarters") < 1 then
```

Comment:
Ok added a different message for this case (not tested)

Reply:
So if Reebaud has not lost when losing his HQ he might lose in the future unnoticed by the player. This may lead to great frustration. So I really would vote to inform the player instantly if has lost the mission, which would mean to have this check and message in a different coroutine running all the time until the mission is fiished